src/worker.cpp

```diff
@@ -1169,24 +1169,44 @@ types::VPDMapVariant Worker::parseVpdFile(const std::string& i_vpdFilePath)
 
     std::shared_ptr<Parser> vpdParser = std::make_shared<Parser>(i_vpdFilePath,
                                                                  m_parsedJson);
-    types::VPDMapVariant l_parsedVpd = vpdParser->parse();
-
-    // Before returning, as collection is over, check if FRU qualifies for
-    // any post action in the flow of collection.
-    // Note: Don't change the order, post action needs to be processed only
-    // after collection for FRU is successfully done.
-    if (jsonUtility::isActionRequired(m_parsedJson, i_vpdFilePath, "postAction",
-                                      "collection"))
+    try
```

Comment:
Should we extend try block for Parser object creation as well ? so we can run "executePostFailAction" .
As we are introducing throwing an exception in Parser constructor.

Reply:
https://github.com/ibm-openbmc/openpower-vpd-parser/pull/380
raising exception in case of error paths.

Reply:
@branupama That can be extended once we introduce that change in parser class.